### PR TITLE
UITextEntryBox text editing shortcuts into sub functions

### PR DESCRIPTION
**Pull request by JordanKarp**
I'm looking to implement some basic text edit functionality within the UITextEntryBox (cut, copy, paste, etc.) and apply it to buttons, like in a menu bar. The UITextEntryBox has all the functionality already, however it is all in one function _process_keyboard_shortcut_event() and tied to keyboard controls.

This proposed change splits the code for Select All, Cut, and Copy into a more extensible format. This format was modeled after how Paste works, with a _process_paste_event() function to check for the correct keyboard process and handled the consumed event, and another function _do_paste() to execute the shortcut. I've split all three other shortcuts into the same format.

If you like this change, I'll add the tests and the documentation before the merge - I think that should be all of it? Let me know if you have any other thoughts! Thanks!

### CODE REVIEW

**Comment by JordanKarp** [comment]
I can also add this to UITextEntryLine if you like, looks to be identical. 

Probably a case for refactoring but this (and #433) seem like a headache that may not be worth it?

**Review by MyreMylar — APPROVED** [review]
This all looks great to me 👍 

I'm not too fussed about any new tests for what is here as there is no new public API and the current tests are covering this code anyway. I'll leave it open for a bit in case you want to add anything anyway.

Thanks for the PR!

**Comment by JordanKarp** [comment]
I thought I added same functionality to UITextEntryLine, so I'm a bit confused by the failed tests. The only thing thats different, upon review, is the UITextEntryLine has an additional condition `and not self.is_text_hidden` which isn't in UITextEntryBox. I'm indifferent about how to proceed - thanks for your help.

File: pygame_gui/elements/ui_text_entry_line.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 from typing import Union, List, Dict, Optional, Tuple
 
 import pygame
+from pygame.event import Event, post
```
Comment (MyreMylar) [inline pygame_gui/elements/ui_text_entry_line.py:7]:
```suggestion
from pygame import Rect, MOUSEBUTTONDOWN, MOUSEBUTTONUP, BUTTON_LEFT, KEYDOWN, TEXTINPUT
from pygame import KMOD_SHIFT, KMOD_META, KMOD_CTRL, KMOD_ALT, K_a, K_x, K_c, K_v
from pygame import K_LEFT, K_RIGHT, K_UP, K_DOWN, K_HOME, K_END, K_BACKSPACE, K_DELETE, K_RETURN
from pygame import key
from pygame.event import Event, post
```

**Comment by MyreMylar** [comment]
OK, I've fixed up those errors and the tests are passing. Thanks for the PR!

**Pull request merged**